api/http/api_devauth.go

```diff
@@ -254,27 +256,42 @@ func (d *DevAuthApiHandlers) PostDevicesV2Handler(w rest.ResponseWriter, r *rest
 	}
 }
 
-func (d *DevAuthApiHandlers) GetDevicesV2Handler(w rest.ResponseWriter, r *rest.Request) {
-
+func (d *DevAuthApiHandlers) SearchDevicesV2Handler(w rest.ResponseWriter, r *rest.Request) {
 	ctx := r.Context()
-
 	l := log.FromContext(ctx)
 
 	page, perPage, err := rest_utils.ParsePagination(r)
 	if err != nil {
 		rest_utils.RestErrWithLog(w, r, l, err, http.StatusBadRequest)
 		return
 	}
+	fltr := model.DeviceFilter{}
 
-	if err = r.ParseForm(); err != nil {
-		err = errors.Wrap(err, "api: malformed query parameters")
-		rest_utils.RestErrWithLog(w, r, l, err, http.StatusBadRequest)
-		return
-	}
+	switch strings.ToLower(r.Header.Get("Content-Type")) {
+	case "application/json", "":
+		err := r.DecodeJsonPayload(&fltr)
+		if err != nil {
+			err = errors.Wrap(err, "api: malformed request body")
+			rest_utils.RestErrWithLog(w, r, l, err, http.StatusBadRequest)
+			return
+		}
 
-	fltr := model.DeviceFilter{}
-	if err = fltr.ParseForm(r.Form); err != nil {
-		rest_utils.RestErrWithLog(w, r, l, err, http.StatusBadRequest)
+	case "application/x-www-form-urlencoded":
+		if err = r.ParseForm(); err != nil {
+			err = errors.Wrap(err, "api: malformed query parameters")
+			rest_utils.RestErrWithLog(w, r, l, err, http.StatusBadRequest)
+			return
+		}
+		if err = fltr.ParseForm(r.Form); err != nil {
+			rest_utils.RestErrWithLog(w, r, l, err, http.StatusBadRequest)
+			return
+		}
+
+	default:
+		rest_utils.RestErrWithLog(w, r, l, errors.Errorf(
+			"Content-Type '%s' not supported",
+			r.Header.Get("Content-Type"),
+		), http.StatusBadRequest)
```

Comment:
what about responding with 415 Unsupported Media Type

Reply:
Actually, this code is unreachable because of the `rest.ContentTypeCheckerMiddleware` but fixed it regardless.